docs/src/tutorial/pure.rst

```diff
@@ -311,15 +311,27 @@ declare types of variables in a Python 3.6 compatible way as follows:
 
 There is currently no way to express the visibility of object attributes.
 
-Cython does not support the full range of annotations described by PEP-484.
-For example it does not currently understand features from the ``typing`` module
-such  as ``Optional[]`` or typed containers such as ``List[str]``. This is partly
-because some of these type hints are not relevant for the compilation to
+``typing`` Module
+^^^^^^^^^^^^^^^^^
+
+Support for the full range of annotations described by PEP-484 is not yet
+complete. Cython 3 currently understands the following features from the
+``typing`` module:
+
+* ``Optional[tp]``, which is interpreted as ``tp or None``;
+* typed containers such as ``List[str]``, which is interpreted as ``list``. The
```

Comment:
do these nest e.g. `List[Optional[str]]`?

Reply:
`List[Optional[Str]]` will probably just end up interpreted as `list` - there's nothing useful that Cython can do with the "element type".

`Optional[List[Str]]` should work (i.e. `list or None`).

I'm not sure I've actually tested the nesting combinations though.